generate/convert.go

```diff
@@ -235,16 +236,23 @@ func (g *generator) convertType(
 	if typ.Elem != nil {
 		// Type is a list.
 		elem, err := g.convertType(
-			namePrefix, typ.Elem, selectionSet, options, queryOptions)
+			namePrefix, typ.Elem, isStructField, selectionSet, options, queryOptions)
 		return &goSliceType{elem}, err
 	}
 
 	// If this is a builtin type or custom scalar, just refer to it.
 	def := g.schema.Types[typ.Name()]
+
 	goTyp, err := g.convertDefinition(
 		namePrefix, def, typ.Position, selectionSet, options, queryOptions)
 
-	if options.GetPointer() {
+	if g.getWeakReference(options, isStructField, def) {
+		if options.Omitempty == nil {
+			oe := true
+			options.Omitempty = &oe
```

Comment:
Mutating this seems a bit awkward.  Can we instead just put this in where we call `GetOmitempty` as well?  That will also help with the comment below I suspect.

Reply:
Agreed, but I played with a few different options here but didn't find anything better.  GetOmitempty is called in a few spots, and we'd have to pass the Config and field definition into it and then duplicating the logic of the `getStructReference` check.

The updated check I've came up with here is a bit cleaner IMO, so see what you think

Reply:
Yeah, fair enough!